apollo-federation/src/sources/connect/mod.rs

```diff
@@ -56,6 +56,7 @@ impl Display for ConnectId {
 #[derive(Debug)]
 pub(crate) struct ConnectFederatedQueryGraph {
     subgraphs_by_name: IndexMap<NodeStr, ValidFederationSubgraph>,
+    source_data: IndexMap<ConnectId, models::Connector>,
```

Comment:
we should make this an Arc right? https://apollograph.slack.com/archives/C06HSR9LS15/p1715605220878099?thread_ts=1715356366.684389&cid=C06HSR9LS15

Reply:
We'll need to review this further as interior mutability will add extra complexity.